test/test_model.py

```diff
@@ -3791,5 +3791,90 @@ def test_repr(self):
         )
 
 
+class TestCloudSpec(unittest.TestCase):
+    def setUp(self) -> None:
+        self.credential = {
+            'auth-type': 'certificate',
+            'attrs': {
+                'client-cert': 'foo',
+                'client-key': 'bar',
+                'server-cert': 'baz'
+            },
+        }
+
+    def test_init(self):
```

Comment:
For what it's worth, if the class becomes a dataclass, we probably don't need this test.